src/status.cpp

```diff
@@ -0,0 +1,77 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "dbops.h"
+#include "status.h"
+
+#include <mio.h>
+
+#include "exceptions.h"
+
+namespace ddb
+{
+
+    void statusIndex(Database *db)
+    {
+
+        const fs::path directory = rootDirectory(db);
+
+        auto q = db->query("SELECT path,mtime,hash FROM entries");
+
+        std::set<std::string> checkedPaths;
+
+        while (q->fetch())
+        {
+            io::Path relPath = fs::path(q->getText(0));
+            auto p = directory / relPath.get(); // TODO: does this work on Windows?
+            Entry e;
+
+            auto path = p.generic_string();
+
+            checkedPaths.insert(path);
+
+            if (exists(p))
+            {
+                if (checkUpdate(e, p, q->getInt64(1), q->getText(2)))
+                {
+                    std::cout << "Modified: " << path << std::endl;
```

Comment:
In anticipation for API integration, this should be invoked as a callback (see `ddb share`) instead of being printed directly to stdout.

Reply:
Addressed in 1cadd7f